crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+            auth_jwt: arg.user_jwt.clone(),
+            auth_seed: arg.user_seed.clone(),
+            lattice_prefix: arg.lattice_id.to_string(),
+            js_domain: arg.js_domain.clone(),
+            ..Default::default()
+        };
+
+        // Setting the auction_timeout to a large value may trigger timeouts in distant code.
+        //
+        // - host configuration rpc_timeout
+        // - `LatticeControllerSender`s which have embedded RPC clients w/ internal timeouts
+        // - httpserver/messaging provider(s) which expect a certain response speed from calling actors
+        //
+        // Since auctions will *wait* until the auction_timeout to do operations like gathering hosts,
+        // we must manually ensure this value is unlikely to cause timeouts.
+        let host_data = wasmcloud_provider_sdk::load_host_data().map_err(|e| {
+            ProviderInvocationError::Provider(format!("failed to load host data: {e}"))
+        })?;
+        if host_data
+            .default_rpc_timeout_ms
+            .is_some_and(|v| v < config.timeout_ms)
+        {
+            warn!(
+                host_rpc_timeout_ms = host_data.default_rpc_timeout_ms,
+                auction_timeout = config.auction_timeout_ms,
+                "host default RPC timeout < auction timeout, operations that rely on auctions are likely to time out"
+            );
+        }
+
+        self.get_connections()
+            .await
+            .put_config(&arg.lattice_id, config)
+            .await;
+
+        Ok(CtlOperationAck {
+            accepted: true,
+            error: "".to_string(),
+        })
+    }
+
+    /// Sets registry credentials, storing them in the cache
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn set_registry_credentials(
+        &self,
+        _ctx: Context,
+        arg: SetRegistryCredentialsRequest,
+    ) -> ProviderInvocationResult<()> {
+        let client = self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id)
+            .await?;
+        let mut hm = HashMap::new();
+        if let Some(ref c) = arg.credentials {
+            for (k, v) in c {
+                hm.insert(
+                    k.to_string(),
+                    interface_client::RegistryCredential {
+                        password: v.password.clone(),
+                        token: v.token.clone(),
+                        username: v.username.clone(),
+                        registry_type: "".to_string(),
+                    },
+                );
+            }
+        }
+
+        client
+            .put_registries(hm)
+            .await
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?;
+        Ok(())
+    }
+
+    /// Auction a provider on the lattice
+    #[instrument(level = "debug", skip(self, _ctx), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
```

Comment:
Change to `skip_all`, especially since this is logging the entire auction request right now

Reply:
- [ ] Fix `skip_all` for `instrument` call